### PR TITLE
MNT: use unordered bulk inserts

### PR DESCRIPTION
**Pull request by klauer**
Minor change which could (?) allow parallel inserts with mongo. Indices should take care of the ordering upon insert.

### CODE REVIEW

**Comment by arkilic** [comment]
Have you checked whether this is faster @klauer ?

**Comment by klauer** [comment]
I haven't done any significant testing with this. It probably should be investigated further prior to merging.

File: filestore/core_v0.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def datum_factory():
                          datum_kwargs=dict(d_kwargs))
             yield datum
 
-    bulk = col.initialize_ordered_bulk_op()
```
Comment (tacaswell) [inline filestore/core_v0.py]: I would leave v0 alone.

**Comment by tacaswell** [comment]
Does this actually make it faster?

**Comment by arkilic** [comment]
Straight from mongo docs: 
```
Ordered vs Unordered Operations

Bulk write operations can be either ordered or unordered.

With an ordered list of operations, MongoDB executes the operations serially. If an error occurs 
during the processing of one of the write operations,
 MongoDB will return without processing any remaining write operations in the list.
With an unordered list of operations, MongoDB can execute the operations in parallel, 
but this behavior is not guaranteed.
 If an error occurs during the processing of one of the write operations, MongoDB 
will continue to process remaining write operations in the list.
```

**Comment by tacaswell** [comment]
That last line is probably why it was `ordered_insert`.  Does this raise on failure?

**Comment by arkilic** [comment]
It will raise on failure. I have seen mdstore raise bulk write errors

**Comment by tacaswell** [comment]
What causes those failures?

**Comment by arkilic** [comment]
Me getting curious and killing the server

**Comment by tacaswell** [comment]
I see.

**Comment by licode** [comment]
Is insertMany function helpful?
https://docs.mongodb.com/manual/reference/method/db.collection.insertMany/#db.collection.insertMany

**Comment by tacaswell** [comment]
@arkilic why are we using the bulk object like this?  https://api.mongodb.com/python/current/examples/bulk.html#bulk-write-operations makes it look like `insert_many` would be simpler as it looks like the `bulk` objects are meant for mixed operations, but we are only doing inset.

**Comment by arkilic** [comment]
As you might see the ```New in version 2.6``` tag above the insert_many(), this is a new feature. Until a few weeks ago, we were maintaining a variety of mongo instance from 2.0 to 3.2.6 There are tons of new features out there, especially after the latest pymongo rewrite. We gotta keep updating as we go along

**Comment by tacaswell** [comment]
The bulk objects are new in 2.7

**Pull request merged**